lib/errors.coffee

```diff
@@ -1,165 +1,956 @@
+###*
+# @namespace errors
+###
+
 TypedError = require('typed-error')
 
-exports.NotFound = class NotFound extends TypedError
+###*
+# An invalid device type error
+# @class ResinInvalidDeviceType
+# @protected
+# @memberof errors
+#
+# @param {String} type - the invalid device type
+###
+exports.ResinInvalidDeviceType = class ResinInvalidDeviceType extends TypedError
+	constructor: (@type) ->
+		###*
+		# @name type
+		# @type String
+		# @memberof errors.ResinInvalidDeviceType
+		# @instance
+		# @constant
+		###
+
+		###*
+		# @name message
+		# @type String
+		# @memberof errors.ResinInvalidDeviceType
+		# @instance
+		# @constant
+		# @default Invalid device type: this.type
+		###
+		@message = "Invalid device type: #{@type}"
```

Comment:
You should change this to `super("Invalid device type: #{@type}")` so that you call the `typed-error` constructor (which handles stack trace construction and the like) - when you have no explicit constructor then it defaults to just calling the parent constructor with all arguments.